### ISSUE TITLE
Child loggers will not emit on parent after addStream()

### ISSUE THREAD

**Issue by thanpolas** [trentm/node-bunyan#479]
I'm using Bunyan for the purposes of an npm library. I have set it up so as to [provide child-loggers for each module](https://github.com/waldophotos/kafka-avro/blob/master/lib/log.lib.js#L26), and then expose the root logger.

When a stream is attached to the root logger using `addStream()` it will not receive any messages logged by the child loggers.

You can reproduce with the following snippet:

```js
var rootLogger = bunyan.createLogger({
  name: 'root',
  level: 'info',
  // Mute logger
  streams: [],
});

var childLog = rootLogger.child({module: 'child-module'});

rootLogger.addStream({
  type: 'stream',
  stream: process.stdout,
}, 'debug');

childLog.debug('Hello there');
childLog.info('Hello there');
childLog.warn('Hello there');
childLog.error('Hello there');

console.log('should have logged');
```

**Comment by thanpolas** [comment]
@trentm can you please confirm this is a bug and not bad-usage? Are there any practical workarounds to my use case so I can get unblocked by this? (provide logs from my library).

**Comment by trentm** [comment]
@thanpolas Sorry for the delay in responding.

As I mentioned in another issue you opened, the intended way to structure libraries is to allow them to take a Bunyan logger instance (created by whatever application or script is using the library), and log on that. This gives the application responsibility (and control) for setting the output streams for logging.
